Yoast/wordpress-seo
# Add several importers for SEO plugins

## Summary

This PR can be summarized in the following changelog entry:

* Add import functionality for [Premium SEO Pack](https://wordpress.org/plugins/premium-seo-pack/).
* Add import functionality for [Smartcrawl SEO](https://wordpress.org/plugins/smartcrawl-seo/).
* Add import functionality for [Squirrly SEO](https://wordpress.org/plugins/squirrly-seo/).
* Add import functionality for [Platinum SEO Pack](https://wordpress.org/plugins/platinum-seo-pack/).
* Add import functionality for [SEO Framework](https://wordpress.org/plugins/autodescription/).
* Add import functionality for [Greg's High Performance SEO](https://wordpress.org/plugins/gregs-high-performance-seo/).
* Add import functionality for [WP Meta SEO](https://wordpress.org/plugins/wp-meta-seo/).
* Improve social data import for wpSEO.de.

## Relevant technical choices:

* Had to work around #9249 in `WPSEO_Plugin_Importer::maybe_save_post_meta`, once #9249 is fixed, we can make this code cleaner again.
* Rewrote the detect mechanism because some plugins use very weird meta keys.
* Importer of Premium SEO Pack extends the Squirrly importer since they use the same (AWFUL) database scheme.

## Test instructions

This PR can be tested by following these steps:

For every plugin that this pull request adds import functionality for:
* Setup one or more posts with the SEO plugin.
* Run `Import`, see that it detects the plugin's data.
* Check data after import.
* Run `Clean`.
* See that data is gone.

## Quality assurance


Fixes #5387
Fixes #5780
Fixes #6316
Fixes #6323 
Fixes #7014
Fixes #7152
Fixes #7279 
Fixes #9025 
Fixes #9270
